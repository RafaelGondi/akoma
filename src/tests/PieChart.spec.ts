import { describe, test, expect } from 'vitest';
import PieChart from '../components/PieChart.vue';
import { shallowMount } from '@vue/test-utils';

const mockedData =[
	{
		name: 'Ecocardiograma',
		datasets: [
			{
				label: 'Exames realizados',
				data: [120, 220, 180, 320, 150],
			},
		]
	},
];

const mockedColors = ['teal', 'violet', 'amber', 'blue', 'red'];

describe('PieChart', () => {
	test('renders correctly', async () => {
		const wrapper = shallowMount(PieChart, {
			props: {
				labels: ['Janeiro','Fevereiro','MarÃ§o', 'Abril', 'Maio'],
				variant: 'green',
				data: mockedData,
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	test('it accepts valid colors', () => {
		const validator = PieChart.props.colors.validator;
		const invalidColors = [...mockedColors, 'silver'];
		expect(validator(mockedColors)).toBe(true);
		expect(validator(invalidColors)).toBe(false);
	});
});
