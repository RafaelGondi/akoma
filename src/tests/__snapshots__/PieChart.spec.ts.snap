// Vitest Snapshot v1

exports[`PieChart > renders correctly 1`] = `
VueWrapper {
  "__app": {
    "_component": [Object],
    "_container": <div … />,
    "_context": [Object],
    "_instance": [Object],
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": [Object],
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "componentVM": {
    "addDataSetNames": [Function],
    "chartOptions": [Object],
    "colors": [Array],
    "computedBackgroundColors": [Array],
    "data": [Array],
    "deleteFirstTwoColors": false,
    "generateBackgroundColor": [Function],
    "hasOwnProperty": [Function],
    "isColorsSet": false,
    "labels": [Array],
    "localChartData": [Object],
    "localLabels": [Array],
    "mergeChartDataNoSelect": [Function],
    "palete": [Function],
    "paleteBuilder": [Function],
    "palletColors": [Array],
    "removeFirstTwoElements": [Function],
    "sassColorVariables": [Object],
    "setColors": [Function],
    "variant": "green",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <span
    data-v-218ded1e=""
  >
    <div … />
  </span>,
}
`;
